src/game/WorldSession.cpp

```diff
@@ -102,17 +99,16 @@ WorldSession::~WorldSession()
         LogoutPlayer(true);
 
     /// - If have unclosed socket, close it
-    if (m_Socket)
-    {
-        m_Socket->CloseSocket();
-        m_Socket->RemoveReference();
-        m_Socket = nullptr;
-    }
+    if (!m_Socket->IsClosed())
+        m_Socket->Close();
 
     ///- empty incoming packet queue
-    WorldPacket* packet = nullptr;
-    while (_recvQueue.next(packet))
+    while (!m_packetQueue.empty())
```

Comment:
You understood me correctly the first time as well, I asked multiple questions at the same time.
Possibly it might have been bugged before, the odds are very slim that it will cause a crash so it might have stayed undetected.